### PR TITLE
[core] add support for more Alert props

### LINKED ISSUE #517: Alert doesn’t have canEscapeKeyClose nor canReturnKeyConfirm
It would be cool if the alert component would have a `canEscapeKeyClose` prop like the dialog component has. Also having `canReturnKeyConfirm` would be doubly cool.

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- `Alert`: add `canEscapeKeyCancel`, `canOutsideClickCancel`, `onClose`, `transitionDuration` props
- `onClose` is single alternative to separate `onCancel`/`onConfirm` props
  - event handlers are all optional now, and handler-less usage is supported
- tweaked warnings, added warnings for new `can*Cancel` props
- __did not add `canEnterKeyConfirm`__ (see #517) because it's not easy to implement (key events require focus, blah blah), and is already supported by simply `tab`-ing onto a button and pressing `enter` or `space`

### CODE REVIEW

File: packages/core/src/common/errors.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ const ns = "[Blueprint]";
 export const CLAMP_MIN_MAX = ns + ` clamp: max cannot be less than min`;
 
 export const ALERT_WARN_CANCEL_PROPS = ns + ` <Alert> cancelButtonText and onCancel should be set together.`;
+export const ALERT_WARN_CANCEL_ESCAPE_KEY = ns + ` <Alert> canEscapeKeyCancel enabled without onCancel handler.`;
+export const ALERT_WARN_CANCEL_OUTSIDE_CLICK = ns + ` <Alert> canOutsideClickCancel enbaled without onCancel handler.`;
```
Comment (invliD) [inline packages/core/src/common/errors.ts]: These are misleading given `onClose` is also ok.

File: packages/core/src/components/alert/alert.tsx [hunk 2]
```diff
@@ -45,54 +66,93 @@ export interface IAlertProps extends IProps {
...
+     * - clicking the confirm button
+     * - focusing on the confirm button and pressing `enter` or `space`
      */
-    onConfirm(e: React.MouseEvent<HTMLButtonElement>): void;
+    onConfirm?(evt?: React.SyntheticEvent<HTMLElement>): void;
+
+    /**
+     * Handler invoked when the Alert is confirmed or canceled; see `onConfirm` and `onCancel` for more details.
+     * First argument is `true` if confirmed, `false` otherwise.
+     * This is an alternative to defining separate `onConfirm` and `onCancel` handlers.
```
Comment (adidahiya) [inline packages/core/src/components/alert/alert.tsx]: why support 3 handlers when one will do? either remove this (leave the callbacks as-is) or remove the existing ones and only support `onClose`

File: packages/core/src/common/errors.ts (hunk 1 above)
Comment (giladgray) [reply packages/core/src/common/errors.ts]: oh yeah gotta update language

File: packages/core/src/components/alert/alert.tsx (hunk 2 above)
Comment (giladgray) [reply packages/core/src/components/alert/alert.tsx]: `TagInput` also has three callbacks

Comment (adidahiya) [reply packages/core/src/components/alert/alert.tsx]: alright fine 🤷‍♂️

**Pull request merged**